rq/worker.py

```diff
@@ -591,6 +592,13 @@ def dequeue_job_and_maintain_ttl(self, timeout):
                 break
             except DequeueTimeout:
                 pass
+            except redis.exceptions.ConnectionError as conn_err:
+                self.log.error('Could not connect to Redis instance: %s '
+                               'Retrying...', conn_err)
+                time.sleep(connection_wait_time)
+                connection_wait_time *= self.exponential_backoff_factor
```

Comment:
Let's limit the `connection_wait_time` to 60 seconds max. So it'll try every 1, 2, 4, 8 ... up to 60 seconds.